core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -450,6 +452,15 @@ public ViewMetadata build() {
           ViewProperties.VERSION_HISTORY_SIZE,
           historySize);
 
+      if (!PropertyUtil.propertyAsBoolean(
+          properties,
+          ViewProperties.REPLACE_DROP_DIALECT_ALLOWED,
+          ViewProperties.REPLACE_DROP_DIALECT_ALLOWED_DEFAULT)) {
+        for (int i = Math.max(1, startingViewVersionsSize); i < versions.size(); i++) {
+          checkIfDialectIsDropped(versions.get(i - 1), versions.get(i));
```

Comment:
This check is overbroad. This check should not cause a view to be completely broken just because a different client put the view into a bad state. Instead, this check should only run to validate the changes made by the caller.

This also makes a bad assumption about view versions. Consecutive view version numbers have no meaning. The current view version may not be the highest number, since the current version ID is tracked separately.

This PR needs to validate that when the current view version changes, there are no dropped dialects.

Reply:
indeed this was a bad assumption on my end. We need to check against the view versions that have been added by the client, which I've changed now and also added a test so that things fail correctly when the view version id is being set by the client